packages/wonder-blocks-modal/src/components/modal-backdrop.tsx

```diff
@@ -34,119 +34,125 @@ type Props = {
  * and adding an `onClose` prop that will call `onCloseModal`. If an
  * `onClose` prop is already provided, the two are merged.
  */
-export default class ModalBackdrop extends React.Component<Props> {
-    componentDidMount() {
+/**
+ * Returns an element specified by the user
+ */
+const getInitialFocusElement = (
+    node: HTMLElement,
+    initialFocusId: string | undefined,
+): HTMLElement | null => {
+    if (!initialFocusId) {
+        return null;
+    }
+
+    // eslint-disable-next-line import/no-deprecated
+    return ReactDOM.findDOMNode(
+        node.querySelector(`#${initialFocusId}`),
+    ) as any;
+};
+
+/**
+ * Returns the first focusable element found inside the Dialog
+ */
+const getFirstFocusableElement = (node: HTMLElement): HTMLElement | null => {
+    // get a collection of elements that can be focused
+    const focusableElements = findFocusableNodes(node);
+
+    if (!focusableElements) {
+        return null;
+    }
+
+    // if found, return the first focusable element
+    return focusableElements[0];
+};
+
+/**
+ * Returns the dialog element
+ */
+const getDialogElement = (node: HTMLElement): HTMLElement => {
+    // If no focusable elements are found,
+    // the dialog content element itself will receive focus.
+    // eslint-disable-next-line import/no-deprecated
+    const dialogElement: HTMLElement = ReactDOM.findDOMNode(
+        node.querySelector('[role="dialog"]'),
+    ) as any;
+    // add tabIndex to make the Dialog focusable
+    dialogElement.tabIndex = -1;
```

Comment:
If an element with `role=dialog` can't be found, [findDOMNode](https://18.react.dev/reference/react-dom/findDOMNode#returns) returns null. And trying to set `tabIndex` on `null` would throw an error! I think if we didn't cast the return value to `any`, then TS would remind us to check the return value first to see if it is null!

As an alternative, I wonder if a ref could be used instead to access the dialog element!

Note: This logic is pre-existing so this issue isn't new!